probe-rs/src/architecture/riscv/communication_interface.rs

```diff
@@ -364,13 +361,22 @@ impl RiscvCommunicationInterface {
         self.enabled_harts & (1 << hart) != 0
     }
 
+    /// Assert the target reset
+    pub fn target_reset_assert(&mut self) -> Result<(), DebugProbeError> {
+        // TODO: Assert sequence if it returns true
+        self.dtm.target_reset_assert()?;
+        Ok(())
+    }
+
     /// Deassert the target reset.
     pub fn target_reset_deassert(&mut self) -> Result<(), DebugProbeError> {
-        self.dtm.target_reset_deassert()
+        // TODO: Deassertion sequence if it returns true
+        self.dtm.target_reset_deassert()?;
+        Ok(())
     }
 
-    /// Read the targets IDCODE.
-    pub fn read_idcode(&mut self) -> Result<u32, DebugProbeError> {
+    /// Read the targets IDCODE if applicable.
+    pub fn read_idcode(&mut self) -> Result<Option<u32>, DebugProbeError> {
```

Comment:
Is this actually necessary? What is the idcode used for?

Reply:
I think we should remove this from here, the `probe-rs info` can just use the JTAG interface directly to get the IDCODE.

Reply:
In the end I did not figure out how to get to raw JTAG interface at the place where the IDCODE is. Instead I have reworded the documentation. It is now an id which is supposed to help the chip selector figure out what kind of chip it is.